File: flatform/src/main/java/nam/tran/flatform/di/PreferenceModule.java
package nam.tran.flatform.di;

import javax.inject.Singleton;

import dagger.Binds;
import dagger.Module;
import nam.tran.flatform.local.IPreference;
import nam.tran.flatform.local.Preference;

@SuppressWarnings("unused")
@Module
public abstract class PreferenceModule {

    @Binds
    @Singleton
    abstract IPreference bindPreference(Preference preferenceProvider);
}

File: data/src/main/java/nam/tran/data/di/PreferenceModule.kt
package nam.tran.data.di

import dagger.Binds
import dagger.Module
import nam.tran.data.local.IPreference
import nam.tran.data.local.Preference
import javax.inject.Singleton

@Module
abstract class PreferenceModule {

    @Binds
    abstract fun providePreference(preferenceProvider: Preference): IPreference
}
